core/js/src/typespecs/prompt.ts

```diff
@@ -0,0 +1,122 @@
+import { z } from "zod";
+
+export const roleSchema = z.enum([
+  "system",
+  "user",
+  "assistant",
+  "function",
+  "tool",
+  "model",
+]);
+export type Role = z.infer<typeof roleSchema>;
+
+export const functionCallSchema = z.object({
+  name: z.string(),
+  arguments: z.string(),
+});
+
+const toolCallSchema = z.object({
+  id: z.string(),
+  function: z.object({
+    arguments: z.string(),
+    name: z.string(),
+  }),
+  type: z.literal("function"),
+});
+
+export const messageSchema = z.object({
+  content: z.string().default(""),
+  role: roleSchema,
+  name: z.string().optional(),
+  function_call: z.union([z.string(), functionCallSchema]).optional(),
+  tool_calls: z.array(toolCallSchema).optional(),
+});
+
+export const promptBlockDataSchema = z.union([
+  z.object({
+    type: z.literal("completion"),
+    content: z.string().default(""),
```

Comment:
Are you sure these defaults will actually get "applied" if someone sends the data through a REST API? If not, maybe we should apply the defaulting logic in one place ubiquitously.